src/Pages/Products/index.js

```diff
@@ -1,83 +1,254 @@
+      const wishListProducts = await addProductsToWishList(product);
+      productDispatch({
+        type: 'ADD_TO_WISHLIST',
+        payload: wishListProducts,
+      });
+      alert(`${product.title} has been added to wishlist`);
+    } else {
+      const wishListProducts = await removeProductsFromWishList(product);
+      productDispatch({
+        type: 'ADD_TO_WISHLIST',
+        payload: wishListProducts,
+      });
+      alert(`${product.title} removed from wishList`);
+    }
+  });
+
   return (
-    <div class='product-container'>
+    <div className='product-container'>
       <aside>
         <ProductFilterComp className='prod-filter-heading'>
           <h4>FILTERS</h4>
-          <button className='btn-primary'>Clear</button>
+          <button className='btn-primary' onClick={handleReset}>
+            Clear
+          </button>
         </ProductFilterComp>
 
         {/* PRICE FILTER */}
         <ProductFilterComp className='prod-filter'>
           <h4 className='prod-filter-type'>PRICE</h4>
-          <input type='range' />
+          <input
+            type='range'
+            max='100000'
+            min='0'
+            step='1000'
+            value={filterState.priceRange}
+            onChange={(e) => {
+              filterDispatch({
+                type: 'RANGE',
+                payload: Number(e.target.value),
+              });
+            }}
+          />
         </ProductFilterComp>
 
         {/* CATEGORY FILTER */}
         <ProductFilterComp className='prod-filter'>
           <h4 className='prod-filter-type'>CATEGORY</h4>
-          {['Fiction', 'Tutorials', 'Non-Fiction'].map((productLabel) => {
-            return <ProductFilter label={productLabel} type='checkbox' />;
-          })}
+          {['Fiction', 'Tutorial', 'Non-Fiction', 'Best-Sellers'].map(
+            (productLabel) => {
+              return (
+                <ProductFilter
+                  key={productLabel}
+                  label={productLabel}
+                  type='checkbox'
+                  onChange={(e) => {}}
+                  checked={
+                    productLabel.toLocaleLowerCase() === param.productName
+                      ? true
+                      : false
+                  }
+                />
+              );
+            }
+          )}
         </ProductFilterComp>
 
         {/* RATING FILTER */}
         <ProductFilterComp className='prod-filter'>
           <h4 className='prod-filter-type'>RATING</h4>
           {[
-            '4 stars & above',
-            '3 stars & above',
-            '2 stars & above',
-            '1 stars & above',
-          ].map((productLabel) => {
-            return <ProductFilter label={productLabel} type='radio' />;
+            { label: '4 stars & above', rateType: 4 },
+            { label: '3 stars & above', rateType: 3 },
+            { label: '2 stars & above', rateType: 2 },
+            { label: '1 stars & above', rateType: 1 },
```

Comment:
Suggestion: you can extract rating and category arrays to separate files and map here.